malaria-app-android/src/main/java/com/peacecorps/malaria/ui/play/badge_screen/BadgeScreenPresenter.java

```diff
@@ -0,0 +1,84 @@
+package com.peacecorps.malaria.ui.play.badge_screen;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+
+import com.peacecorps.malaria.R;
+import com.peacecorps.malaria.data.AppDataManager;
+import com.peacecorps.malaria.ui.base.BasePresenter;
+import com.peacecorps.malaria.ui.play.badge_screen.BadgeScreenContract.BadgeMvpPresenter;
+import com.peacecorps.malaria.ui.play.badge_screen.BadgeScreenContract.BadgeMvpView;
+
+/**
+ * Created by Anamika Tripathi on 24/7/18.
+ */
+public class BadgeScreenPresenter<V extends BadgeMvpView> extends BasePresenter<V> implements BadgeMvpPresenter<V> {
+    BadgeScreenPresenter(AppDataManager manager, Context context) {
+        super(manager, context);
+    }
+
+    @Override
+    public void attachView(V view) {
+        super.attachView(view);
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+    }
+
+    @Override
+    public int getUserScore() {
+        return getDataManager().getUserScore();
+    }
+
+    @Override
+    public String getUserName() {
+        return getDataManager().getUserName();
+    }
+
+    //get badge based on medication  score
+    @Override
+    public void selectUserBadge() {
+        Drawable badgeDrawable= null;
+        int userScore = getDataManager().getUserScore();
+        //set badge according to score
+        if(userScore<2){
+            badgeDrawable=getContext().getResources().getDrawable(R.drawable.y_b1);
```

Comment:
I have pointed this out before as well, good coding practice is to get context and check if not null, getResources, if not null, and then get into all if cases otherwise, handle null context/resources separately.